src/plots/plots.js

```diff
@@ -1102,3 +1129,43 @@ plots.graphJson = function(gd, dataonly, mode, output, useDefaults) {
 
     return (output === 'object') ? obj : JSON.stringify(obj);
 };
+
+/**
+ * Modify a keyframe using a list of operations:
+ *
+ * @param {array of objects} operations
+ *      Sequence of operations to be performed on the keyframes
+ */
+plots.modifyFrames = function(gd, operations) {
+    var i, op;
+    var _frames = gd._frameData._frames;
+    var _hash = gd._frameData._frameHash;
+
+    for(i = 0; i < operations.length; i++) {
+        op = operations[i];
+
+        switch(op.type) {
+            case 'rename':
+                var frame = _frames[op.index];
+                delete _hash[frame.name];
+                _hash[op.name] = frame;
+                break;
+            case 'replace':
+                frame = op.value;
+                _frames[op.index] = _hash[frame.name] = frame;
```

Comment:
And here, needs lookup table cleanup if the name has changed, I think. (though the only current use is when the name *doesn't* change—but it's by index, so that should be handled to prevent headaches down the road)